graphene/src/layers/vector/vector_shape.rs

```diff
@@ -106,4 +111,110 @@ impl VectorShape {
 	fn to_local_space(&self, point: kurbo::Point) -> DVec2 {
 		self.transform.transform_point2(DVec2::from((point.x, point.y)))
 	}
+
+	/// TODO: remove kurbo from below implementations
```

Comment:
So currently it looks like our flow is kurbo -> VectorShape -> kurbo (via From) does that sum up the general life cycle here? Would it make sense to move all of this back out of VectorShape since it technically should be fully separated from any external lib? I saw you deleted the shape_adapter.rs which originally housed this, maybe we can reform some external adapter pattern?

Reply:
The reason I moved `impl From<&VectorShape> for BezPath`  out of `shape_adaptor.rs` is because it seemed guaranteed that if a crate used  `shape_adaptor.rs` it must also be using `vector_shape.rs`. And `vector_shape.rs` isn't too big yet.

Right now many parts of the editor do flow like kurbo -> VectorShape -> kurbo, but as we create our own implementations of things like `apply_affine`, VectorShape won't have to rely on kurbo. In `shape_layer.rs` we use kurbo for shape prototypes, `kurbo::Rect` for example. These prototypes are things that VectorShape should have eventually, `VectorShape::Rect()`. In other parts of the editor there is a weird mix between glam types and kurbo types, but most of the reason there are kurbo types is to be compatible with functions that we will want to implement for VectorShape anyway. So once VectorShape doesn't have to rely on kurbo the rest of editor shouldn't have to either. 

Other parts of the editor rely on kurbo more heavily for its Bezier stuff, boolean_ops and the pen_tool for example. But this reliance is necessary unless we create our own representation of Bezier curves. Because this functionality does math on, and modifies, the actual curve we definitely need to be able to modify VectorShapes as Bezier curves. Will we use kurbo's traits and types or create our own?

Then the last place we use kurbo is to go from Bezier to SVG, but I think the SVG stuff is going to be replaced by WebGL anyways?

Reply:
For now, we want to limit the leveraging of kurbo to the systems that absolutely require it. I know that the pen tool, as it stands, actually doesn't need the curve representation. However, I know the boolean operations do and some future changes to the pen tool will as well, so we will need to implement curve functionality asap as well. I am fine with moving it into VectorShape for now, with the knowledge we will eventually move it out.